src/com/hms/flexyosisoftconnector/DataPoster.java

```diff
@@ -7,14 +7,40 @@
    private static final int maxDataPointsPerPost = 200;
    private boolean shouldRun;
 
-   public void run()
+   public DataPoster (int communicationType) {
+	   switch(communicationType) {
+       case OSIsoftConfig.omf:
+    	   Logger.LOG_INFO("OMF support enabled");    	   
+           break;
+       case OSIsoftConfig.piwebapi:
+    	   Logger.LOG_INFO("Using connection settings for piwebapi 2018 and earlier");
+           break;
+       default:
+           Logger.LOG_ERR("Immpossible to reach here, checks done in config class.");

```

Comment:
errors like this are vague as the line number isnt logged. maybe reword to "Internal Errror: Communication type is invalid."